### PR TITLE
Rotation gate unitary

### PR DESCRIPTION
**Pull request by sduquemesa**
**Context:**
The rotation gate unitary is currently being calculated using the Choi isomorphisms which generates `nan` for angles of value zero where the identity matrix should be returned.

**Description of the Change:**
The rotation gate being diagonal in the Fock basis is easy to implement with better speed performance. This PR implements the `U` method of the rotation gate overriding the default way of calculating the unitaries of Gaussian transforms.

**Benefits:**
Fast computation of the rotation gate Fock representation which avoids invalid numerical outcomes.

**Possible Drawbacks:**
None

### CODE REVIEW

File: mrmustard/lab/gates.py [hunk 1]
```diff
@@ -177,6 +178,47 @@ def __init__(
...
+        num_args = (
+            args.shape[0] if len(args.shape.as_list()) > 0 else 1
+        )  # number or arguments given to the gate
+        angles = np.zeros((num_modes,))
+        if num_args == 1:
+            # one arg for all modes
+            angles[modes] = args
+        elif num_args == len(modes):
+            # an arg for each mode
+            angles[modes] = args
```
Comment (elib20) [inline mrmustard/lab/gates.py:215]: Should this be angles = args? Otherwise there is no difference from the outcome of the preceding if statement

**Review by elib20 — COMMENTED** [review]
Nice! A couple small questions

File: mrmustard/lab/gates.py [hunk 2]
```diff
@@ -177,6 +178,47 @@ def __init__(
...
+        elif num_args == len(modes):
```
[the remark this answers is not in the document]
Comment (sduquemesa) [reply mrmustard/lab/gates.py:216]: yes!

File: mrmustard/lab/gates.py (hunk 1 above)
Comment (sduquemesa) [reply mrmustard/lab/gates.py:215]: yes! Thanks for spotting this

**Review by elib20 — APPROVED** [review]

**Review by ziofil — APPROVED** [review]
I made a couple of changes, hopefully tests are all ok

**Pull request merged**